src/pkg/cli/update.go

```diff
@@ -0,0 +1,55 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/DefangLabs/defang/src/pkg/term"
+)
+
+func Update(ctx context.Context) error {
+	// Find path to the current executable to determine how to update
+	ex, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	term.Debugf(" - Executable: %s\n", ex)
+
+	ex, err = filepath.EvalSymlinks(ex)
+	if err != nil {
+		return err
+	}
+	term.Debugf(" - Evaluated: %s\n", ex)
+
+	if strings.Contains("brew/", ex) {
+		term.Info(" * Updating defang via Homebrew...")
+		return exec.CommandContext(ctx, "brew", "upgrade", "defang").Run()
+	}
+
+	if strings.Contains("/nix/store/", ex) {
```

Comment:
I think these args are backwards[0], and we might want to use `HasPrefix` instead of `Contains`:

```go
if strings.HasPrefix(ex, "/nix/store/") {
```

[0] https://pkg.go.dev/strings#Contains